### PR TITLE
[Merged by Bors] - fix: Let controller watch AuthenticationClasses

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

leftover of #434

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/main.rs [hunk 1]
```diff
@@ -74,7 +75,9 @@ async fn main() -> anyhow::Result<()> {
                 watch_namespace.get_api::<TrinoCluster>(&client),
                 watcher::Config::default(),
             );
-            let cluster_store = cluster_controller.store();
+            let cluster_store_1 = Arc::new(cluster_controller.store());
+            let cluster_store_2 = cluster_store_1.clone();
```
Comment (maltesander) [inline rust/operator-binary/src/main.rs:79]: Name it `catalog_cluster_store` and `authentication_class_cluster_store`?

**Review by maltesander — APPROVED** [review]
LGTM!

File: rust/operator-binary/src/main.rs (hunk 1 above)
Comment (sbernauer) [reply rust/operator-binary/src/main.rs]: Honestly I don't care :shrug:

**Pull request closed without merging**